cli/cli_app.go

```diff
@@ -460,6 +460,7 @@ func checkVersionConstraints(terragruntOptions *options.TerragruntOptions) error
 		terragruntOptions,
 		nil,
 		[]config.PartialDecodeSectionType{config.TerragruntVersionConstraints},
+		false,
```

Comment:
Sanity checks: 

1. Does this mean you can no longer use `locals` in version constraints? I could see this being problematic if someone is trying to manage version constraints across many places with a single local variable somewhere...
1. What other parts of `terragrunt.hcl` do we parse as part of checking version constraints? Could those other parts also be using `run_cmd`? Should we instead update this code to explicitly _only_ parse version constraints and no other parts of the file?
1. Whatever we decide, we should update the docs accordingly.

Reply:
@amnk Bump ^

Reply:
@brikis98 So it actually works like that already. At least kinda like that. In our code we have a [decodeList](https://github.com/amnk/terragrunt/blob/amnk/issues-1427/config/config_partial.go#L20) option, to pass what exactly should be parsed, and when we check versions - we [specify that](https://github.com/amnk/terragrunt/blob/amnk/issues-1427/cli/cli_app.go#L462). But `locals` and `include` are always parsed. This is probably because as you said - someone can use locals in version constraints. My code added ability to switch off `locals` parsing.

I initially thought that Terragrunt wants to follow Terraform, which [does not support interpolation syntax in versions](https://github.com/hashicorp/terraform/issues/15896). Imho this is the most stable approach, and this will help us avoid creating lots of special cases in the code.